### PR TITLE
exodii weaponmaster

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
People tie too much stuff to hub 01; i don't like it
also i don't like that there is too little npcs in rocky butte
#### Describe the solution
add weaponmaster that partially handles weapon and armor trade; maybe in future they could do more, but man i'd rather yell at code for a week than make npc
#### Describe alternatives you've considered
remove it's merchant abilities, just making them vibin' here
#### Testing
image
image
image
#### Additional context
perfect counterpart to untranslateable dialogues of Rubik

### CODE REVIEW

File: data/json/npcs/exodii/exodii_weaponmaster.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  {
+    "type": "mutation",
+    "id": "EXODII_BODY_9",
+    "flags": [ "Exodii_Body" ],
+    "name": { "str": "Exodii cyborg framework - style 9" },
+    "points": 0,
+    "visibility": 9,
+    "ugliness": 7,
+    "mixed_effect": true,
+    "description": "Your entire body is covered by a rectangle frame, that has numerous supportive arms and manipulators, with decent stability.  Price for this is that the weight of entire frame is enormous, which causes user to either use dedicated tracked treads to move around, or stay immovable for a large amount of time.",
```
Comment (AudBobb) [inline data/json/npcs/exodii/exodii_weaponmaster.json:69]: >weight of entire frame -> >weight of the entire frame
>which causes user -> >which causes the user
>or stay immovable -> >or stay immobile

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
+[
+  {
+    "id": "TALK_WEAPONMASTER_INTRO_1",
+    "type": "talk_topic",
+    "//": "mute by choise, and deaf most of the time (don't like to turn on the hearing cbm unless for job)",
+    "dynamic_line": "&What you see looks like the back of a mechanical spider, with at least a dozen crude but polished metal arms going from the enormously wide spine and shoulders.  Spine of a creature, meanwhile, sits in an awkward-looking chair, doing something on a wide, heavily lighted workbench.  Creature seems to be busy with something.",
```
Comment (AudBobb) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:6]: >polished metal arms going from -> >polished metal arms coming from 

>Spine of a creature 
I think you are referencing the mechanical spider's spine here in which case this should be
>The spine of the creature

>wide, heavily lighted 
This should probably be changed to
>wide, brightly lit
but if you wanted you could leave heavily in place of brightly. Lighted is also correct, and more popular in British English, so this one is subjective.

>Creature seems to be -> >The creature seems to be

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    "responses": [
+      { "text": "&Wait a bit.", "effect": "start_trade", "topic": "TALK_WEAPONMASTER_INTRO_2A" },
+      { "text": "Hello?", "topic": "TALK_WEAPONMASTER_INTRO_2B" }
+    ]
+  },
+  {
+    "id": "TALK_WEAPONMASTER_INTRO_2A",
+    "type": "talk_topic",
+    "dynamic_line": "&It doesn't react at your presence at all.",
```
Comment (AudBobb) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:15]: >It doesn't react at your -> >It doesn't react to your

**Comment by John-Candlebury** [comment]
Oh cool stuff.

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    "dynamic_line": "&It seems to not hear you at all.",
+    "responses": [
+      { "text": "&Wait for longer.", "effect": "start_trade", "topic": "TALK_WEAPONMASTER_INTRO_3" },
+      { "text": "Hello?", "topic": "TALK_WEAPONMASTER_INTRO_3" }
+    ]
+  },
+  {
+    "id": "TALK_WEAPONMASTER_INTRO_3",
+    "type": "talk_topic",
+    "dynamic_line": "&A few seconds later, they seem to finish whatever they were busy with and slowly turn their chair to you.  Their front looks no less exotic than their back; the abundance of metal arms looks even more extraordinary next to the clear lack of any legs.  What you initially thought to be a chair is simply a rotating platform, sturdy enough to support the weight of the entire body of a creature.  Few cameras, that seems to be a replacement for eyes, look at you motionlessly, while the forehead of the creature has the surprisingly familiar sign \"IX\" painted on.",
```
Comment (AudBobb) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:33]: >turn their chair to you -> >turn their chair towards you
>entire body of a creature -> entire body of the creature

>Few cameras, that seems to be a  -> >A number of cameras, which seem to be a

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    "responses": [ { "text": "Who-…  Wha-…  What are you?", "topic": "TALK_WEAPONMASTER_INTRO_4" } ]
+  },
+  {
+    "id": "TALK_WEAPONMASTER_INTRO_4",
+    "type": "talk_topic",
+    "dynamic_line": "&Still completely mute, it shows you something you completely missed: The entire wall behind the workbench was full of various weapons: some as big as modern rifles, a few as large as machineguns installed on aircraft, and one that looked like dismantled tank turrets.  Melee weapons were not left behind, from moderate-sized swords and clubs to warhammers twice the human height and what you can describe only as \"boots for mecha\".  Then it pointed to the sign near you that showed the carved pictograms of coins, some unknown for you sign, and then a bullet - all things considered, it is either a weaponmaster or a weapon merchant.",
```
Comment (AudBobb) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:39]: This whole thing needs to be restructured

Still completely mute, it shows you something you had initially missed. The entire wall behind the workbench is full of various weapons; some as big as modern rifles, a few as large as the machine guns installed on aircraft, and one that looks like a dismantled tank turret.  Melee weapons were not forgotten, from moderately-sized swords and clubs to war hammers twice the height of a human, and what you can describe only as \"boots for mecha\".  Then it points to a sign near you that shows the carved pictograms of coins, some unknown to you, next to a pictogram of a bullet - all things considered, this creature is either a weapons master or a weapons merchant.

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    "dynamic_line": "&They don't move and do not say anything.",
+    "responses": [ { "text": "Do you understand me?", "topic": "TALK_WEAPONMASTER_INTRO_6" } ]
+  },
+  {
+    "id": "TALK_WEAPONMASTER_INTRO_6",
+    "type": "talk_topic",
+    "dynamic_line": "&They clearly do not understand you.  It seems they do not hear you either.",
+    "responses": [
+      {
+        "text": "&You pull off your backpack, and start to pull different stuff you deem worthy of trading.  The weaponmaster finally looks interested.",
```
Comment (AudBobb) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:54]: >pull different stuff you deem -> >pull out items you deem

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        "topic": "TALK_NONE",
+        "effect": [ { "npc_first_topic": "TALK_EXODII_WEAPONMASTER" }, "start_trade" ]
+      }
+    ]
+  },
+  {
+    "id": "TALK_EXODII_WEAPONMASTER",
+    "type": "talk_topic",
+    "dynamic_line": [
+      "&Weaponmaster seems to tinkering something, but turned their chair to you as you approached.",
```
Comment (AudBobb) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:64]: The Weapons Master seems to be tinkering on something, but turns their chair to you as you approach.

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+      "&Weaponmaster seems to clean some weapon, but as you approach them, they call someone to pick a gun, and turn to you.",
```
Comment (AudBobb) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:65]: The Weapons Master seems to be cleaning some sort of weapon, but as you approach them, they call out to someone to take the weapon, and turn to you.

I believe your meaning of this >they call someone to pick a gun< is the weapon master is calling someone to take the weapon they were working on back to the armory or something along those lines, so I corrected the grammar under that assumption.

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+      "&Weaponmaster seems to polish some armor, but turn to you the second you enter the room."
```
Comment (AudBobb) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:66]: The Weapons Master seems to be polishing some armor, but they turn towards you the second you enter the room.

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    ],
+    "responses": [
+      {
+        "text": "&With familiar move, you take off different valuables on the trader counter, proposing a trade.",
```
Comment (AudBobb) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:70]: With a familiar move, you place a variety of valuables onto the counter, proposing a trade.

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json (hunk 2 above)
Comment (PatrikLundell) [reply data/json/npcs/exodii/exodii_weaponmaster_talk.json:6]: choise -> choice

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json (hunk 8 above)
Comment (PatrikLundell) [reply data/json/npcs/exodii/exodii_weaponmaster_talk.json:65]: I don't think the Exodii would use sound, but rather radio, especially if sound receivers are typically disabled.

**Comment by PatrikLundell** [comment]
If Rubik won't trade certain stuff he'd probably need dialog to direct you to the appropriate trader(s), with the message that they don't understand Anglic, so you'd have to communicate using gestures.
One thing you could do is to use Rubik's dialog to tell you the names of them so you could have one dialog start where you call the name of the one you address, and one for a general "Hello there". You could also have introductions, one for when you know the name of your counterpart (only presenting yourself), and one for when you make a general call and your counterpart introduces itself first and you thereafter (automatically, unless you want a rude alternative, which I wouldn't provide if I tried to write it).

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json (hunk 8 above)
Comment (AudBobb) [reply data/json/npcs/exodii/exodii_weaponmaster_talk.json:65]: So maybe try 

>The Weapons Master seems to be cleaning some sort of weapon, but as you approach them, they signal to someone to take the weapon, and turn to you.

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 11]
```diff
@@ -0,0 +1,76 @@
+[
+  {
+    "id": "TALK_WEAPONMASTER_INTRO_1",
+    "type": "talk_topic",
+    "//": "mute by choise, and deaf most of the time (don't like to turn on the hearing cbm unless for job)",
+    "dynamic_line": "&What you see looks like the back of a mechanical spider, with at least a dozen crude but polished metal arms going from the enormously wide spine and shoulders.  Spine of a creature, meanwhile, sits in an awkward-looking chair, doing something on a wide, heavily lighted workbench.  Creature seems to be busy with something.",
+    "responses": [
+      { "text": "&Wait a bit.", "effect": "start_trade", "topic": "TALK_WEAPONMASTER_INTRO_2A" },
```
Comment (Uwuewsky) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:8]: Narrative responses use a style like this:
https://github.com/CleverRaven/Cataclysm-DDA/blob/5d320508149b28da05fc4d45e4a8f2d1c7547ef6/data/json/npcs/robofac/robofac_intercom/robofac_intercom_missions.json#L592
```suggestion
      { "text": "[Wait a bit.]", "effect": "start_trade", "topic": "TALK_WEAPONMASTER_INTRO_2A" },
```

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 12]
```diff
@@ -61,13 +61,13 @@
     "id": "TALK_EXODII_WEAPONMASTER",
     "type": "talk_topic",
     "dynamic_line": [
-      "&Weaponmaster seems to tinkering something, but turned their chair to you as you approached.",
-      "&Weaponmaster seems to clean some weapon, but as you approach them, they call someone to pick a gun, and turn to you.",
-      "&Weaponmaster seems to polish some armor, but turn to you the second you enter the room."
+      "&The Weaponmaster seems to be tinkering on something, but turns their chair to you as you approach.",
+      "&The Weaponmaster seems to be cleaning some sort of weapon, but as you approach them, they signal to bystanding worker to take the weapon, and turn to you.",
```
Comment (AudBobb) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:65]: they signal to bystanding worker -> they signal to a bystanding worker

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 13]
```diff
@@ -2,17 +2,17 @@
   {
     "id": "TALK_WEAPONMASTER_INTRO_1",
     "type": "talk_topic",
-    "//": "mute by choise, and deaf most of the time (don't like to turn on the hearing cbm unless for job)",
-    "dynamic_line": "&What you see looks like the back of a mechanical spider, with at least a dozen crude but polished metal arms going from the enormously wide spine and shoulders.  Spine of a creature, meanwhile, sits in an awkward-looking chair, doing something on a wide, heavily lighted workbench.  Creature seems to be busy with something.",
+    "//": "mute by choice, and deaf most of the time (don't like to turn on the hearing cbm unless for job)",
+    "dynamic_line": "&What you see looks like the back of a mechanical spider, with at least a dozen crude but polished metal arms coming from the enormously wide spine and shoulders.  Spine of a creature, meanwhile, sits in an awkward-looking chair, doing something on a wide, heavily lighted workbench.  Creature seems to be busy with something.",
```
Comment (AudBobb) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:6]: Spine of a creature, meanwhile, -> The spine of the creature, meanwhile,
Creature seems to -> The creature seems to

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 14]
```diff
@@ -0,0 +1,76 @@
+[
+  {
+    "id": "TALK_WEAPONMASTER_INTRO_1",
+    "type": "talk_topic",
+    "//": "mute by choice, and deaf most of the time (don't like to turn on the hearing cbm unless for job)",
+    "dynamic_line": "&What you see looks like the back of a mechanical spider, with at least a dozen crude but polished metal arms coming from the enormously wide spine and shoulders.  The spine of the creature, meanwhile, sits in an awkward-looking chair, doing something on a wide, heavily lighted workbench.  The creature seems to be busy with something.",
```
Comment (Maleclypse) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:6]:
```suggestion
    "dynamic_line": "&What you see looks like the back of a mechanical spider, with at least a dozen crude but polished metal arms coming from the enormously wide spine and shoulders.  The spine of the creature, meanwhile, sits in an awkward-looking chair, manipulating tools and geegaws on a wide, brightly lit workbench.  The creature seems to be busy with something.",
```

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+    "dynamic_line": "&It seems to not hear you at all.",
+    "responses": [
+      { "text": "[Wait in silence.]", "topic": "TALK_WEAPONMASTER_INTRO_3" },
+      { "text": "Hello!?", "topic": "TALK_WEAPONMASTER_INTRO_3" }
+    ]
+  },
+  {
+    "id": "TALK_WEAPONMASTER_INTRO_3",
+    "type": "talk_topic",
+    "dynamic_line": "&A few seconds later, they seem to finish whatever they were busy with and turn their chair towards you.  Their front looks no less exotic than their back; the abundance of metal arms looks even more extraordinary next to the clear lack of any legs.  What you initially thought to be a chair is simply a rotating platform, sturdy enough to support the weight of the entire body of the creature.  A number of cameras, which seem to be a replacement for eyes, look at you motionlessly, while the forehead of the creature has the surprisingly familiar sign \"IX\" painted on.",
```
Comment (Maleclypse) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:33]:
```suggestion
    "dynamic_line": "&A few seconds later, they seem to finish whatever they were busy with and turn their chair towards you.  Their front looks no less exotic than their back; the abundance of metal arms are even more extraordinary next to the clear lack of any legs.  What you initially thought to be a chair is simply a rotating platform, sturdy enough to support the weight of the entire body of the creature.  A number of cameras, which seem to be a replacement for eyes, look at you motionlessly, while the forehead of the creature has the surprisingly familiar sign \"IX\" painted on.",
```

**Comment by Maddremor** [comment]
Does the weaponmaster have to be an NPC, or are our talking monster systems robust enough for a trader?

**Comment by GuardianDll** [comment]
Monsters can't trade afaik

File: data/json/npcs/exodii/exodii_weaponmaster_talk.json [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+    "responses": [ { "text": "Who-…  Wha-…  What are you?", "topic": "TALK_WEAPONMASTER_INTRO_4" } ]
+  },
+  {
+    "id": "TALK_WEAPONMASTER_INTRO_4",
+    "type": "talk_topic",
+    "dynamic_line": "&Still completely mute, it shows you something you had initially missed.  The entire wall behind the workbench is full of various weapons; some as big as modern rifles, a few as large as the machine guns installed on aircraft, and one that looks like a dismantled tank turret.  Melee weapons were not forgotten, from moderately-sized swords and clubs to war hammers twice the height of a human, and what you can describe only as \"boots for mecha\".  Then it points to a sign near you that shows three things: the carved pictograms of coins, some sign you are not familiar with, and a pictogram of a bullet - all things considered, this creature is either a weaponmaster, or a weapon merchant.",
+    "responses": [ { "text": "Woah.  Can i buy one?", "topic": "TALK_WEAPONMASTER_INTRO_5" } ]
```
Comment (Night-Pryanik) [inline data/json/npcs/exodii/exodii_weaponmaster_talk.json:40]:
```suggestion
    "responses": [ { "text": "Woah.  Can I buy one?", "topic": "TALK_WEAPONMASTER_INTRO_5" } ]
```

**Pull request merged**